openapi/payments_api.yaml

```diff
@@ -0,0 +1,535 @@
+      responses:
+        '204':
+          description: No Content
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '409':
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+  /api/v1/payments/{paymentid}/cancel:
+    parameters:
+      - $ref: '#/components/parameters/correlationIdParam'
+    post:
+      summary: cancel payment
+      operationId: cancel-payment
+      parameters:
+        - name: paymentid
+          in: path
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '204':
+          description: No Content
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '409':
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+  /v1/payments/cancel:
+    parameters:
+      - $ref: '#/components/parameters/correlationIdParam'
+    post:
+      summary: cancel payment using idempotency key
+      operationId: cancel-payment-idempotency
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CancelPaymentRequest'
+      responses:
+        '204':
+          description: No Content
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '409':
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+  /v1/integration-test/payments/{paymentid}/reserve:
+    parameters:
+      - $ref: '#/components/parameters/correlationIdParam'
+    post:
+      summary: simulates payment confirmation
+      operationId: simulate-confirmation
+      description: Simulate user swiping and confirming the payment (**available only in sandbox environment**).
+      servers:
+        - url: https://api.sandbox.mobilepay.dk
+          description: This endpoint is available only in sandbox environment
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CancelPaymentRequest'
```

Comment:
Seems like integraton-test/reserve should have a different requestBody

Reply:
fixed